lib/nerves_init_gadget/gadget_devices.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Nerves.InitGadget.GadgetDevices do
+  @moduledoc """
+  Set up the gadget devices with usb_gadget
+  """
+  use GenServer
+
+  @doc false
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def init(_opts) do
+    USBGadget.Builtin.create_rndis_ecm_acm("g")
+    USBGadget.enable_device("g")
+
+    # Set up the bond0 interface across usb0 and usb1.
+    # In the rndis_ecm_acm pre-defined device being used here, usb0 is the
+    # RNDIS (Windows-compatible) device and usb1 is the ECM
+    # (non-Windows-compatible) device.
+    # Since Linux supports both with ECM being more reliable, we set usb1 (ECM)
+    # as the primary, meaning that it will be used if both are available.
+    :os.cmd('ip link set bond0 down')
+    File.write!("/sys/class/net/bond0/bonding/mode", "active-backup")
+    File.write!("/sys/class/net/bond0/bonding/miimon", "100")
+    File.write!("/sys/class/net/bond0/bonding/slaves", "+usb0")
+    File.write!("/sys/class/net/bond0/bonding/slaves", "+usb1")
+    File.write!("/sys/class/net/bond0/bonding/primary", "usb1")
+    :os.cmd('ip link set bond0 up')
```

Comment:
## :os.cmd
I _think_ we could use `nerves_network_interface` to do this correct? Might not be worth the trouble given the current state of the network stack.

## File.write
Having all these `File.write!/2`'s here makes me feel a bit weird. I don't know of a better solution however.

Reply:
Yeah, I'm not opposed to having this bonding functionality built into `nerves_network` or `nerves_network_interface`, but it from scanning `nerves_network_interface`, it wasn't obvious to me how to do it/where to add it. I'm not familiar with the C APIs and these `/sys/class/net` interfaces made it really easy to do this one thing I need to do here.

Maybe we leave it like this for now and migrate it into the other librar(ies) as we have time to get them more fleshed-out and complete?